NetTopologySuite.IO/NetTopologySuite.IO/IO/Streams/FileStreamProvider.cs

```diff
@@ -19,10 +21,10 @@ public FileStreamProvider(string kind, string path, bool validatePath = false)
             if (path == null)
                 throw new ArgumentNullException("path");
 
-#if NET40
+#if HAS_SYSTEM_STRING_ISNULLORWHITESPACE
             if (string.IsNullOrWhiteSpace(path))
 #else
-            if (string.IsNullOrEmpty(path))
+            if (string.IsNullOrEmpty(path) || path.All(Char.IsWhiteSpace))

```

Comment:
I'm assuming this one is not available:

https://msdn.microsoft.com/en-us/library/system.string.isnullorwhitespace(v=vs.110).aspx

LINQ is usually bad for performance but because this is checking a path I'm assuming this happens only once when opening a filestream and won't have that much impact any.

I would just leave this in.